python/metric_pipeline_utils/tex.py

```diff
@@ -0,0 +1,249 @@
+
+    Returns
+    -------
+    measurement : `lsst.verify.Measurement`
+        Measurement of TEx (x=1,2) and associated metadata.
+
+    Notes
+    -----
+    The TEx table below is provided in ``validate_drp``\ 's :file:`metrics.yaml`.
+
+    LPM-17 dated 2011-07-06
+
+    Specification:
+        Using the full survey data, the E1, E2, and EX residual PSF ellipticity
+        correlations averaged over an arbitrary FOV must have the median
+        less than TE1 for theta <= 1 arcmin, and less than TE2 for theta >= 5 arcmin.
+
+    The residual ellipticity correlations vary smoothly so it is sufficient to
+    specify limits in these two angular ranges. On 1 arcmin to 5 arcmin scales,
+    these residual ellipticity correlations put LSST systematics a factor of a
+    few below the weak lensing shot noise, i.e., statistical errors will
+    dominate over systematics. On larger scales, the noise level imposed by
+    nature due to shot noise plus cosmic variance is almost scale-independent,
+    whereas the atmospheric contribution to systematics becomes negligible.
+    Therefore the specifications on 5 arcmin scales apply to all larger scales
+    as well (as per section 2.1.1). On scales larger than the field of view,
+    sources of systematic error have less to do with the instrumentation than
+    with the operations (due to the seeing distribution), software, and algorithms.
+
+    ========================= ====== ======= =======
+    PSF Ellipticity Residuals     Specification
+    ------------------------- ----------------------
+                       Metric Design Minimum Stretch
+    ========================= ====== ======= =======
+            TE1 ()             2e-5    3e-5   1e-5
+            TE2 (%)            1e-7    3e-7   5e-8
+            TEF (%)             15      15     10
+            TE3 ()             4e-5    6e-5   2e-5
+            TE4 ()             2e-7    5e-7   1e-7
+    ========================= ====== ======= =======
+
+
+    Table 27: These residual PSF ellipticity correlations apply to the r and i bands.
+    """
+
+    matches = matchedDataset.matchesBright
+
+    datums = {}
+    datums['D'] = Datum(quantity=D, description="Separation distance")
+
+    radius, xip, xip_err = correlation_function_ellipticity_from_matches(matches, verbose=verbose)
+    datums['radius'] = Datum(quantity=radius, description="Correlation radius")
+    datums['xip'] = Datum(quantity=xip, description="Correlation strength")
+    datums['xip_err'] = Datum(quantity=xip_err, description="Correlation strength uncertainty")
+    datums['bin_range_operator'] = Datum(quantity=bin_range_operator, description="Bin range operator string")
+
+    operator = ThresholdSpecification.convert_operator_str(bin_range_operator)
+    corr, corr_err = select_bin_from_corr(radius, xip, xip_err, radius=D, operator=operator)
+    quantity = np.abs(corr) * u.Unit('')
+    return Measurement(metric, quantity, extras=datums)
```

Comment:
As above, I'm uncomfortable having this here and partially ported in the run method of the measure task.